volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -105,40 +105,62 @@ def get_sections(self):
             yield attr
 
     def get_symbols(self):
-        if symbols.symbol_table_is_64bit(self._context, self.get_symbol_table().name):
+        """Get module symbols"""
+        if symbols.symbol_table_is_64bit(self._context, self.get_symbol_table_name()):
             prefix = "Elf64_"
         else:
             prefix = "Elf32_"
         elf_table_name = intermed.IntermediateSymbolTable.create(
-            self.context,
-            self.config_path,
+            self._context,
+            self._context.modules["kernel"].config_path,
             "linux",
             "elf",
             native_types=None,
             class_types=elf.class_types,
         )
 
         syms = self._context.object(
-            self.get_symbol_table().name + constants.BANG + "array",
+            self.get_symbol_table_name() + constants.BANG + "array",
             layer_name=self.vol.layer_name,
             offset=self.section_symtab,
             subtype=self._context.symbol_space.get_type(
                 elf_table_name + constants.BANG + prefix + "Sym"
             ),
             count=self.num_symtab + 1,
         )
+
         if self.section_strtab:
             for sym in syms:
-                sym.set_cached_strtab(self.section_strtab)
-                yield sym
+                try:
+                    sym_offset = self.section_strtab + sym.st_name
+                    sym_name = self._context.layers[self.vol.layer_name].read(
+                        sym_offset, 512
+                    )  # 512 is the value of KSYM_NAME_LEN
+                except exceptions.PagedInvalidAddressException:
+                    continue
+
+                # Stop at first null byte (strtab is a null terminated strings list)
+                sym_name = sym_name.split(b"\x00")[0].decode("latin-1")
```

Comment:
Hey @Abyss-W4tcher - Just a minor one, more a function that that might be helpful for you to know than anything else. For the this name part, you might be able to use the `pointer_to_string` utility rather than doing the split for the null byte yourself. It also handles errors nicely for you as well if there are weird characters. 

https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/objects/utility.py#L23

Reply:
Hi, thanks for pointing this out to me ! Excluding the use of "self._context" which isn't great, as discussed earlier, could you tell me if it is the right way to do it : 

```python
try:
  pointer_to_sym_offset = self._context.object("pointer", sym_offset)
except PagedInvalidAddressException: 
 continue 

sym_name = utility.pointer_to_string(pointer_to_sym_offset, 512)
if sym_name != "": 
 [...]
```

Thanks

Reply:
Sorry, I gave the wrong advice earlier, `self._context` is the right way to do it, given there is no other means of getting the object's context, but it is private, so reaching into another object's `_context` would be bad.  Hope that makes sense...

Reply:
Yes, if you have an integer, you should be able to cast it to any other volatility type.  Pointers can be instantiated even on invalid addresses though, so best to check that integer against the layer using, I think, `is_valid` first...